File: Builder/python-source/Chef.py
from base.Burger import Burger
from builders.BurgerBuilder import BurgerBuilder


class Chef(object):
    def __init__(self):
        self.builder: BurgerBuilder

    def set_builder(self, builder: BurgerBuilder):
        self.builder = builder

    def cook(self) -> Burger:
        self.builder.reset()
        self.builder.add_low_bread()
        self.builder.add_vegetable()
        self.builder.add_meat()
        self.builder.add_vegetable()
        self.builder.add_high_bread()
        return self.builder.get_burger()

File: JavaSource/src/main/java/ru/akhcheck/patterns/builder/Chef.java
package ru.akhcheck.patterns.builder;

import ru.akhcheck.patterns.builder.base.Burger;
import ru.akhcheck.patterns.builder.builders.BurgerBuilder;

public class Chef {
    private BurgerBuilder builder;

    public Burger Cook() {
        return (
            builder
                .reset()
                .addLowBread()
                .addMeat()
                .addVegetable()
                .addHighBread()
                .getBurger()
        );
    }

    public void setBuilder(BurgerBuilder builder) {
        this.builder = builder;
    }
}
